### PR TITLE
Introduce waitForLoaderToBeRemoved util

### LINKED ISSUE #34422: Introduce waitForLoaderToBeRemoved util
See: https://github.com/metabase/metabase/pull/34316#discussion_r1345402161

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #34422

### Description

- Introduce `waitForLoadingSpinnerToDisappear` and replace all existing _equivalent_ code with it ([almost _equivalent_](https://metaboat.slack.com/archives/C505ZNNH4/p1684753566358179?thread_ts=1684751522.480859&cid=C505ZNNH4))
- Remove all usages of `waitForElementToBeRemoved`
- Replace all loading spinner queries by text/regexp with queries by `data-testid`
- Consolidate `@testing-library/react` & `__support__/ui` imports in the modified files

Why:
- we should stop using `waitForElementToBeRemoved` (see [Slack discussion](https://metaboat.slack.com/archives/C505ZNNH4/p1684753502335459?thread_ts=1684751522.480859&cid=C505ZNNH4) and [PR discussion](https://github.com/metabase/metabase/pull/34272#discussion_r1342527087))
- 1 consistent way of waiting for spinner to disappear
- 1 consistent way of querying loading spinner using `data-testid`
- fewer LOC on average (in some places equivalent assertion took 5 LOC - now 1 LOC +/- import) and definitely fewer LOC in test code (i.e. excluding imports formatting) - improved vertical distance 

### How to verify

Only unit tests (should) have been modified.
There should be no flakes or errors.

### CODE REVIEW

File: frontend/test/__support__/ui.tsx
Comment (kamilmielnik) [inline frontend/test/__support__/ui.tsx:198]: Here's the util

Comment (uladzimirdev) [reply frontend/test/__support__/ui.tsx:198]: nit: add a comment why it's needed with the reference to a bug in RTL

File: frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.unit.spec.tsx
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.unit.spec.tsx:262]: comment above is related to the loading

```suggestion
```

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.unit.spec.tsx
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.unit.spec.tsx:471]: here we test that loading exists and then disappears from the dom. and it's the logic of `waitForElementToBeRemove` actually. Do you think we should check both in the util? (that element first appears then disappears)

**Review by uladzimirdev — APPROVED** [review]
I like the idea overall, many thanks for this. 

My main concern is whether we should check that the loading element first exists in the dom and then disappears

**Review by uladzimirdev — APPROVED** [review]
I like the idea overall, many thanks for this. 

My main concern is whether we should check that the loading element first exists in the dom and then disappears

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.unit.spec.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.unit.spec.tsx:471]: > Do you think we should check both in the util?

The util was created to avoid exactly that.

True, if both assertions are needed (there aren't that many places like that though), then `waitForElementToBeRemoved` would be the right choice, although I prefer explicit code, like here.

----

If we ever get async issues with this assertion, we could add another util, like this:
```tsx
await waitForLoadingSpinnerToApppear();
await waitForLoadingSpinnerToDisappear();
```

**Comment by kamilmielnik** [comment]
> My main concern is whether we should check that the loading element first exists in the dom and then disappears

You're right, we will lose the assertion that spinner was displayed.
My take is that a great majority of existing `waitForElementToBeRemoved` calls didn't mean to check that - all that was required was to wait for the spinner to be gone, so the actual testing can start/continue.

@uladzimirdev Please let me know if you think it's worth keeping this assertion.

I'll wait for at least one more approval before merging.

File: frontend/test/__support__/ui.tsx
Comment (kamilmielnik) [reply frontend/test/__support__/ui.tsx:210]: 2f2e87e91423ce8b3f4365fb1db18aef62113757

**Review by ranquild — APPROVED** [review]
Maybe `waitForLoaderToBeRemoved`? E.g. use the `mantine` component name and `toBeRemoved` prefix from react testing library

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.unit.spec.tsx
Comment (uladzimirdev) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.unit.spec.tsx:471]: no objections

**Comment by uladzimirdev** [comment]
@kamilmielnik from my perspective it's fine

**Comment by kamilmielnik** [comment]
> Maybe `waitForLoaderToBeRemoved`? E.g. use the `mantine` component name and `toBeRemoved` prefix from react testing library

Good idea! 18536489fae13ca9a61f0981d99b45150ad9663b

**Review by iethree — APPROVED** [review]
LGTM, thanks for taking this on!

**Pull request merged**
